src/main/java/io/vertx/openapi/contract/impl/PathFinder.java

```diff
@@ -92,4 +98,17 @@ boolean testSegments(String[] pathSegments, String[] pathTemplateSegments) {
     }
     return true;
   }
+
+  private int getNumSegmentsMatching(String[] pathSegments, String[] pathTemplateSegments) {
```

Comment:
Maybe it is possible to combine `testSegments` and `getNumSegmentsMatching` by returning the number of matching segments.

```
  // VisibleForTesting
  int testSegments(String[] pathSegments, String[] pathTemplateSegments) {
   int numPerfectMatches = 0;
    for (int i = 0; i < pathTemplateSegments.length; i++) {
      String templateSegment = pathTemplateSegments[i];
      if (templateSegment.contains("{")) {
        // valid segment
      } else if (templateSegment.equals(pathSegments[i])) {
        numPerfectMatches += pathTemplateSegments.length - i;
      } else {
        return -1;
      }
    }
    return numPerfectMatches;
  }
```

The downside of this is, that the method becomes more complex and we should add a JavaDoc which explains it a little bit, but I guess it will improve the performance a little bit.

What do you think?

Reply:
In case we don't combine the two methods, we should have a small unit test for `getNumSegmentsMatching` like the one for `testSegments`

Reply:
I have no issues combing the two methods. I only kept them separated as I didn't want to break the signature of ```testSegments``` . But looking at it now I realize PathFinder is a private class. I'll make the update to combine the two later tonight!